x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -514,6 +525,20 @@ public boolean checkFeature(Feature feature) {
         return allowed;
     }
 
+    /**
+     * Marks a licensed feature as <em>on by configuration</em>.
+     * By default {@link #getFeatureUsage()} method returns the last time {@link #checkFeature(Feature)} was called.
+     * One this method is called, the specified feature will be marked as "always in use",and {@link #getFeatureUsage()} method will return
+     * the current time ("now"), instead of the last time the feature was checked.
+     */
+    public synchronized boolean setFeatureActive(Feature feature, String... identifiers) {
+        boolean allowed = checkFeature(feature);
+        if (allowed && isTrackableFeature(feature)) {
```

Comment:
A few interesting points here:
* Do we need a pairing `unset` method? This could be useful if we have any "on-by-configuration" features that can turned on and off on the fly? Or if the license expires or gets deleted or is somehow downgraded, I assume it's the caller's responsibility to call the `unset` method so it's removed from the tracking? Or should unlicensed features be filtered out at `getFeatureUsage` time?
* If the feature is not trackable, should we throw an `assert` error to notify the developer that license is not configured correctly?
* How can we ensure that every "on-by-default" feature (past and future) is indeed registered with this method?